### PR TITLE
feat: expose PDF page rendering API across all language bindings

### PR DESCRIPTION
**Pull request by tobocop2**
I hope you don't mind one last PR from me for a while. I know this is a large one but it's largely docs, tests and bindings. Normally, I'd say it makes sense to do multiple PR's, but all of these things go together and I don't think it makes sense to break this up. This is the last gap that will allow me to delegate fully to kreuzberg. 

kreuzberg already renders PDF pages to images internally for its OCR fallback pipeline, but this isn't exposed publicly. I needed this for a project where I send PDF pages to vision LLMs for OCR. Vision LLMs produce way better results than Tesseract on complex layouts, tables, and handwritten notes, but they need page images as input.

Without this API I had to bring in `pypdfium2` separately to rasterize. Being able to rasterize pages and send them to a vision model within one library could simplify my project significantly and enable me to drop that dependency.

### The API

Two functions: `render_pdf_page` for a single page by index, and `PdfPageIterator` for lazy page-by-page iteration. Both return PNG bytes. Available in all 11 language bindings.

### Why this shape

I deliberately kept the API to two things: a single-page function and an iterator. No batch "give me all pages" function — a 500-page PDF at 150 DPI would produce ~1GB of PNGs in memory. The iterator reads the file once, renders one page at a time, and only keeps one PNG in memory. The single-page function covers the case where you already know which page you want.

### What changed

**Core (~130 lines of new Rust logic):** `render_pdf_page_to_png`, `PdfPageIterator`, `encode_png`. Wraps existing `PdfRenderer`.

**C FFI:** `CPageImage`, `CPageIterResult`, `CPdfPageIterator` opaque handle with render/iterator/free functions.

**All 11 language bindings:** single-page render + lazy iterator using idiomatic patterns per language (Python context manager, Go Close(), Java AutoCloseable, C# IDisposable, Elixir Stream, Node PdfPageIterator, Ruby yield block, R callback, PHP Generator).

**Tests:** 3 JSON fixtures generate happy-path render tests for all 11 languages via the e2e generator. Edge case tests (nonexistent file, empty path, out-of-bounds page, negative index, iterator cleanup, early termination) cover all 9 languages with binding-specific test directories (Rust and C are covered by e2e only).

No existing APIs or behavior changed and this is purely additive.

### Usage

```python
from kreuzberg import render_pdf_page, PdfPageIterator

page_0 = render_pdf_page("document.pdf", 0, dpi=200)

with PdfPageIterator("document.pdf", dpi=150) as pages:
    for page_index, png in pages:
        text = vision_model(png)
```

### Testing

Locally verified render tests pass across all 11 languages:

- **E2e (generated):** Python 3, Rust 3, Go 3, Ruby 3, R 9, Elixir 3, Node 3, Java 3, C# 3, C 3, PHP 3
- **Edge cases (9 languages):** Python 8, Node 8, Go 5, Elixir 6, PHP 8, C# 8, Java 8, R 13, Ruby 8

### CI status (fork runs)

Fork CI runs are linked below to supplement local testing.

| Workflow | Status | Notes |
|----------|--------|-------|
| [CI C FFI](https://github.com/tobocop2/kreuzberg/actions/runs/23586604534) | 🟢 pass |  |
| [CI C#](https://github.com/tobocop2/kreuzberg/actions/runs/23586604574) | 🟢 pass |  |
| [CI CLI](https://github.com/tobocop2/kreuzberg/actions/runs/23586604536) | 🟢 pass |  |
| [CI Docker Build & Test](https://github.com/tobocop2/kreuzberg/actions/runs/23586604520) | 🟡 flaky | [flaky on main](https://github.com/kreuzberg-dev/kreuzberg/actions/runs/23557072203) |
| [CI Elixir](https://github.com/tobocop2/kreuzberg/actions/runs/23586604513) | 🟡 pre-existing | [fails on main](https://github.com/kreuzberg-dev/kreuzberg/actions/runs/23557072255) |
| [CI Go](https://github.com/tobocop2/kreuzberg/actions/runs/23586604584) | 🟢 pass |  |
| [CI Java](https://github.com/tobocop2/kreuzberg/actions/runs/23586604523) | 🟡 flaky | tesseract build.rs panic on Windows, flaky |
| [CI Node](https://github.com/tobocop2/kreuzberg/actions/runs/23586604552) | 🟡 pre-existing | [fails on main](https://github.com/kreuzberg-dev/kreuzberg/actions/runs/23557072264) |
| [CI PHP](https://github.com/tobocop2/kreuzberg/actions/runs/23586604563) | 🟢 pass |  |
| [CI Python](https://github.com/tobocop2/kreuzberg/actions/runs/23586604524) | 🟢 pass |  |
| [CI R](https://github.com/tobocop2/kreuzberg/actions/runs/23586604572) | 🟢 pass |  |
| [CI Ruby](https://github.com/tobocop2/kreuzberg/actions/runs/23586604531) | 🟢 pass |  |
| [CI Rust](https://github.com/tobocop2/kreuzberg/actions/runs/23586604539) | 🟡 pre-existing | e2e contract tests fail, not render-related |
| [CI Validate](https://github.com/tobocop2/kreuzberg/actions/runs/23586604533) | 🟡 pre-existing | [fails on main](https://github.com/kreuzberg-dev/kreuzberg/actions/runs/23557072292) |
| [CI WASM](https://github.com/tobocop2/kreuzberg/actions/runs/23586604535) | 🟡 pre-existing | [fails on main](https://github.com/kreuzberg-dev/kreuzberg/actions/runs/23557072264) |

### CODE REVIEW

File: crates/kreuzberg/src/pdf/rendering.rs [hunk 1]
```diff
@@ -64,16 +64,10 @@ impl PdfRenderer<'_> {
         options: &PageRenderOptions,
         password: Option<&str>,
     ) -> Result<DynamicImage> {
-        let document = self.pdfium.load_pdf_from_byte_slice(pdf_bytes, password).map_err(|e| {
-            let err_msg = super::error::format_pdfium_error(e);
-            if (err_msg.contains("password") || err_msg.contains("Password")) && password.is_some() {
```
Comment (tobocop2) [inline crates/kreuzberg/src/pdf/rendering.rs:69]: outside of scope, but made this repeated error handling code reusable.
